openbas-api/src/test/java/io/openbas/rest/exercise/ExerciseApiStatusTest.java

```diff
@@ -263,37 +292,52 @@ void runExerciseAfterPauseTest() throws Exception {
     // --PREPARE--
     ExerciseUpdateStatusInput input = new ExerciseUpdateStatusInput();
     input.setStatus(ExerciseStatus.RUNNING);
-
-    // --EXECUTE--
-    String response =
-        mvc.perform(
-                put(EXERCISE_URI + "/" + PAUSED_EXERCISE.getId() + "/status")
-                    .content(asJsonString(input))
-                    .contentType(MediaType.APPLICATION_JSON)
-                    .accept(MediaType.APPLICATION_JSON))
-            .andExpect(status().is2xxSuccessful())
-            .andReturn()
-            .getResponse()
-            .getContentAsString();
-
-    // --ASSERT--
-    Thread.sleep(2000);
-    List<ExecutableInject> injects = injectHelper.getInjectsToRun();
-    List<Pause> pauses = pauseRepository.findAllForExercise(PAUSED_EXERCISE.getId());
-    Optional<Exercise> exercise =
-        exerciseRepository.findById(JsonPath.read(response, "$.exercise_id"));
-    if (exercise.isPresent()) {
-      Exercise responseExercise = exercise.get();
-      assertEquals(Optional.empty(), responseExercise.getCurrentPause());
+    Instant instantBeforeSetRun = REFERENCE_TIME;
+    Instant instantAfterSetRun = REFERENCE_TIME.plus(1, MINUTES);
+    Clock clock = Clock.fixed(REFERENCE_TIME, ZoneId.of("UTC"));
+
+    try (MockedStatic<Instant> mockedInstant = mockStatic(Instant.class, CALLS_REAL_METHODS)) {
+      try (MockedStatic<Clock> mockedClock = mockStatic(Clock.class, CALLS_REAL_METHODS)) {
+
+        mockedInstant.when(Instant::now).thenReturn(instantBeforeSetRun);
+        // we need this other mock because the production code
+        // inconsistently calls Instant.now() and LocalDateTime.now()
+        mockedClock.when(Clock::systemDefaultZone).thenReturn(clock);
+
+        // --EXECUTE--
+        String response =
+            mvc.perform(
+                    put(EXERCISE_URI + "/" + PAUSED_EXERCISE.getId() + "/status")
+                        .content(asJsonString(input))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn()
+                .getResponse()
+                .getContentAsString();
+
+        // --ASSERT--
+        // NOTE: we are changing the time of Instant.now() here to fast forward in the future
+        mockedInstant.when(Instant::now).thenReturn(instantAfterSetRun);
```

Comment:
question: I don't see any assertion requiring a time comparison. I take it that it's still necessary to mock it somehow, but is there a way to create an assertion around it ?

Reply:
We don't sepcifically assert on times except the expected `exercise_start_date` as discussed here https://github.com/OpenBAS-Platform/openbas/pull/2089#discussion_r1888712580